### PR TITLE
[PT] Translation of chapter 2

### PR DESCRIPTION
**Pull request by johnnv1**
#51

### CODE REVIEW

**Review by lewtun — COMMENTED** [review]
Thank you for translating the whole chapter 🤯 ! This looks great and the only thing missing is to update the `_toctree.yml` file by including these new sections, e.g. please add

```yaml
- title: 2. Using 🤗 Transformers # Translate this
  sections:
  - local: chapter2/1
    title: Introduction # Translate this
  - local: chapter2/2
    title: Behind the pipeline # Translate this
  - local: chapter2/3
    title: Models # Translate this
  - local: chapter2/4
    title: Tokenizers # Translate this
  - local: chapter2/5
    title: Handling multiple sequences # Translate this
  - local: chapter2/6
    title: Putting it all together # Translate this
  - local: chapter2/7
    title: Basic usage completed! # Translate this
  - local: chapter2/8
    title: End-of-chapter quiz # Translate this
    quiz: 2
```

**Comment by johnnv1** [comment]
I now see that this will keep the chapter directory name in English, I will fix this

**Comment by lewtun** [comment]
> just now i see that this will keep the chapter directory name in English, I will fix this

Thanks! I'm just running the CI to test the docs build, but also see we have some code formatting issues. Could you run

```
python -m pip install -r requirements.txt
```

followed by 

```
make style
```

to fix the errors please?

**Comment by lewtun** [comment]
I think we have a small problem with nested quotes "" in the alt text here

```
<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface-course/documentation-images/resolve/main/en/chapter2/full_nlp_pipeline-dark.svg" alt="O pipeline NLP completa: tokenização do texto, conversão para IDs, e inferência através do Transformer e pela "cabeça" do modelo."/>
```

File: chapters/pt/chapter2/8.mdx [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+			text: "Primeiro, o modelo trata de textos e devolve previsões em bruto. O tokenizer, então, trás sentido a estas previsões e as converte de volta ao texto quando necessário.",
+			explain: "O modelo não consegue entender o texto! O tokenizer deve primeiro simbolizar o texto e convertê-lo em IDs para que seja compreensível pelo modelo."
+		},
+		{
+			text: "Primeiro, o tokenizer trata de textos e devolve as identificações (IDs). O modelo lida com estas identificações e produz uma predição, que pode ser algum texto.",
+			explain: "A predição do modelo não pode ser feita de imediato. O tokenizer tem que ser usado para converter a predição de volta ao texto!"
+		},
+		{
+			text: "O tokenizer trata de textos e devolve os IDs. O modelo lida com estes IDs e produz uma predição. O tokenizer pode então ser usado mais uma vez para converter estas previsões de volta para algum texto.",
+			explain: Correto! O tokenizer pode ser usado tanto para a tokenização quanto para a destokenização.",
```
Comment (lewtun) [inline chapters/pt/chapter2/8.mdx:21]: I think we're missing a quote here :)

```suggestion
			explain: "Correto! O tokenizer pode ser usado tanto para a tokenização quanto para a destokenização.",
```

**Comment by lewtun** [comment]
Thanks for iterating - it seems to have worked, but we'll need `chapter1/1.mdx` translated to see the full content on the website. Merging this now 🚀

**Pull request merged**